src/main/java/lotto/domain/LottoGame.java

```diff
@@ -37,62 +52,44 @@ public LottoGame(List<String> ticketTexts) {
         return strings;
     }
 
-    private List<LottoTicket> generateLottoTickets(int ticketCount) {
+    private static List<LottoTicket> generateLottoTickets(int ticketCount) {
         List<LottoTicket> tickets = new ArrayList<>();
         for (int i = 0; i < ticketCount; i++) {
             tickets.add(new LottoTicket());
         }
         return tickets;
     }
 
-    public void doGame(String winText, int bonus) {
-        LottoTicket winTicket = new LottoTicket(winText);
-
-        if (!isValidBonus(winTicket, bonus)) {
-            throw new RuntimeException();
-        }
-
-        this.winCounts = calculateWinCounts(winTicket, bonus);
-        this.winPercent = calculateWinPercent();
-    }
-
-    private boolean isValidBonus(LottoTicket winTicket, int bonus) {
-        return !(winTicket.contains(bonus)
-                || bonus < NumberGenerator.LOTTO_START_NUM
-                || bonus > NumberGenerator.LOTTO_END_NUM);
+    public LottoStatistics doGame(String winText, int bonus) {
+        Map<String, Integer> winCounts = calculateWinCounts(new WinTicket(winText, bonus));
```

Comment:
이왕 만든 `Rank`를 사용하면 어떨까요?

```suggestion
        Map<Rank, Integer> winCounts = calculateWinCounts(new WinTicket(winText, bonus));
```